### PR TITLE
Feat/validator test contract

### PR DESCRIPTION
**Pull request by ratik**

### CODE REVIEW

File: validator_test.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+done
+
+if [ ${#ICA_ADDRESS} != ${ICA_LENGTH[$i]} ]
+then
+    echo "Can't get ICA address"
+    exit
+fi
+echo "ICA address: $ICA_ADDRESS"
+echo ""
+echo "Please send 0.02 ${DENOM[$i]} to $ICA_ADDRESS"
```
Comment (foxpy) [inline validator_test.sh:143]: You probably wanted to write this?
```suggestion
echo "Please send 0.02 ${TARGET_DENOMS[$i]} to $ICA_ADDRESS"
```

**Review by foxpy — CHANGES_REQUESTED** [review]

File: validator_test.sh (hunk 1 above)
Comment (ratik) [reply validator_test.sh]: thanks. I wanted DENOMS bc I want to say like send 0.02 (atom/junox) and not uatom/ujunox

File: contracts/neutron_validator_test/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Neutron Interchain Txs Contract
+
+The contract provides a convenient interface for interacting with [Interchain Txs Module](https://github.com/neutron-org/neutron/tree/master/x/interchaintxs).
+
+The contract responsible for providing sudo endpoint receiving various acknowledgments
```
Comment (foxpy) [inline contracts/neutron_validator_test/README.md:5]: Please reword it into something like "Neutron Test Contract for Validators"

File: contracts/neutron_validator_test/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+    let delegate_msg = MsgDelegate {
+        delegator_address: delegator,
+        validator_address: validator,
+        amount: Some(Coin {
+            denom,
+            amount: amount.to_string(),
+        }),
+    };
+    let mut buf = Vec::new();
+    buf.reserve(delegate_msg.encoded_len());
```
Comment (foxpy) [inline contracts/neutron_validator_test/src/contract.rs:212]:
```suggestion
    let mut buf = Vec::with_capacity(delegate_msg.encoded_len());
```

File: validator_test.sh [hunk 4]
```diff
@@ -0,0 +1,176 @@
+args=("$@")
+## Set up envs
+echo "What network you going to interact:"
+echo "1. CosmosHub"
+echo "2. Juno"
+
+read -N 1 i
+echo $i
```
Comment (foxpy) [inline validator_test.sh:8]: This echoes back user input twice and doesn't wait until user presses Enter (making them scared).
I would rather use `read i` in this case.

**Review by foxpy — CHANGES_REQUESTED** [review]

File: validator_test.sh [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+do
+    ((j--))
+    RES=$(neutrond query wasm contract-state smart ${CONTRACT_ADDRESS} "{\"interchain_account_address_from_contract\":{\"interchain_account_id\":\"${INTERCHAIN_ACCOUNT_ID}\"}}" --chain-id ${NEUTRON_CHAIN_ID} --output json 2>/dev/null)
+    ICA_ADDRESS=$(echo $RES | jq -r '.data | .[0]')
+    if [ ${#ICA_ADDRESS} = ${ICA_LENGTH[$i]} ]
+    then
+	break
+    fi
+    sleep 5
+done
```
Comment (foxpy) [inline validator_test.sh:134]: This loop always fails for me, no matter if I choose Juno or Cosmos Hub :(

Comment (ratik) [reply validator_test.sh:133]: lemme know if this one still an issue

Comment (foxpy) [reply validator_test.sh:133]: nope, now it works

**Review by foxpy — APPROVED** [review]
Great job!

**Review by oopcode — APPROVED** [review]

**Pull request merged**